functions/pipeline/onboarding/__init__.py

```diff
@@ -9,10 +26,106 @@ def main(req: func.HttpRequest) -> func.HttpResponse:
         req_body = req.get_json()
         logging.error(req.get_json())
         url_list = req_body["imageUrls"]
-        url_string = (", ".join(url_list))
     except ValueError:
         print("Unable to decode JSON body")
         return func.HttpResponse("Unable to decode POST body", status_code=400)
 
     logging.error(req_body)
-    return func.HttpResponse("Got body: " + url_string, status_code=200)
\ No newline at end of file
+
+    # Build list of image objects to pass to DAL for insertion into DB.
+    image_object_list = []
+    image_name_list = []
+    
+    # TODO: Add check to ensure image URLs sent by client are all unique.
+
+    for url in url_list:
+        # Split original image name from URL
+        original_filename = url.split("/")[-1]
+        image_name_list.append(original_filename)
+        # Create ImageInfo object (def in db_access.py)
+        # Note: For testing, default image height/width are set to 50x50
+        # TODO: Figure out where actual height/width need to come from
+        image = DB_Access_V2.ImageInfo(original_filename, url, 50, 50)
+        # Append image object to the list
+        image_object_list.append(image)
+
+    logging.info("Now connecting to database...")
+    db_config = DB_Access_V2.DatabaseInfo(os.getenv('DB_HOST', default_db_host), os.getenv('DB_NAME', default_db_name), os.getenv('DB_USER', default_db_user), os.getenv('DB_PASS', default_db_pass))
+    data_access = DB_Access_V2.ImageTagDataAccess(DB_Access_V2.PostGresProvider(db_config))
+    logging.info("Connected.")
+
+    # Create user id
+    user_id = data_access.create_user(DB_Access_V2.getpass.getuser())
+    logging.info("The user id for '{0}' is {1}".format(DB_Access_V2.getpass.getuser(),user_id))
+
+    # Add new images to the database, and retrieve a dictionary ImageId's mapped to ImageUrl's
+    image_id_url_map = data_access.add_new_images(image_object_list,user_id)
+
+    # Print out dictionary for debugging
+    logging.info("Image ID and URL map dictionary:")
+    logging.info(image_id_url_map)
+
+    # Copy over images to permanent blob store and save URLs in a list
+    permanent_url_list = []
+    update_urls_dictionary = {}
+
+    # TODO: Add check to make sure image exists in temp storage before attempting these operations
+    for key, value in image_id_url_map.items():
+
+        # Verbose logging for testing
+        logging.info("Key: " + key)
+        logging.info("Value: " + str(value))
+
+        original_image_url = key
+        original_blob_name = original_image_url.split("/")[-1]
+        file_extension = os.path.splitext(original_image_url)[1]
+        image_id = value
+        new_blob_name = (str(image_id) + file_extension)
+        permanent_storage_path = DESTINATION_CONTAINER_URL + "/" + new_blob_name
+
+        # Verbose logging for testing
+        logging.info("Original image URL: " + original_image_url)

```

Comment:
If you want, as an upskilling exercise we can work together to refactor the blob functionality into a blobClient class and isolate it for better testing. It will also make this main function easier to read.

Reply:
Sure...I don't mind taking a stab at this on my own, but if you want to work together on it we can, might get done more quickly that way